sdk/management/azure/src/main/java/com/azure/management/Azure.java

```diff
@@ -384,51 +301,47 @@ public SdkContext sdkContext() {
 
         @Override
         public Azure withSubscription(String subscriptionId) {
-            return new Azure(restClient, subscriptionId, tenantId, this);
+            profile.withSubscriptionId(subscriptionId);
+            return new Azure(httpPipeline, profile, this);
         }
 
         @Override
-        public Azure withDefaultSubscription() throws CloudException, IOException {
-            if (this.defaultSubscription != null) {
-                return withSubscription(this.defaultSubscription);
-            } else {
-                PagedIterable<Subscription> subs = this.subscriptions().list();
-                if (subs.iterator().hasNext()) {
-                    return withSubscription(subs.iterator().next().subscriptionId());
-                } else {
-                    return withSubscription(null);
-                }
```

Comment:
Keep this in order for default subs. Meanwhile add a with default tenant like it. Since these two could be null in profile. I think it more friendly than all need user to add.

Reply:
Personally I kind of against this code. It might just pick a wrong subscription id and mess things up.

Reply:
em, I'm not sure how many user will use only one subs for every credential. Like me, my auth will always be only one subs.

Ok to remove it as I think it may be many beginner use the code in sample and will not set subs when call the default subs.

Reply:
Yes, file usually only have one, but that is already coded in that file (and we kind of deprecate file for now). Our tenant actually got 10+ subIds.

Maybe we can add e.g. `withSingleSubscription` if requirement arise (but still prefer only do this if there is only 1 subscriptionId avaliable, hence the mock name).

Reply:
No, My auth will always list one subs from azure, since I only add one subs permission to one service principal, it will not list other subs. 

I think we could use `withProfileSubscription` and `withFirstSubscriptionFetched` to distinguish if we want. Sometimes the `withDefaultSubscription` is a little strange.